src/Raven.Server/Documents/Handlers/Admin/Processors/Revisions/AbstractAdminRevisionsHandlerProcessorForDeleteRevisions.cs

```diff
@@ -15,24 +20,57 @@ protected AbstractAdminRevisionsHandlerProcessorForDeleteRevisions([NotNull] TRe
         {
         }
         
-        protected abstract ValueTask DeleteRevisionsAsync(TOperationContext context, string[] documentIds, bool includeForceCreated,
+        protected abstract Task<long> DeleteRevisionsAsync(DeleteRevisionsOperation.Parameters parameters,
             OperationCancelToken token);
 
         public override async ValueTask ExecuteAsync()
         {
-            bool includeForceCreated = RequestHandler.GetBoolValueQueryString("includeForceCreated", required: false) ?? false;
-
-            using (ContextPool.AllocateOperationContext(out TOperationContext context))
+            DeleteRevisionsOperation.Parameters parameters = null;
+            var deletedCount = 0L;
+            Exception ex = null;
+            try
             {
-                var json = await context.ReadForMemoryAsync(RequestHandler.RequestBodyStream(), "admin/revisions/delete");
-                var parameters = JsonDeserializationServer.Parameters.DeleteRevisionsParameters(json);
+                using (ContextPool.AllocateOperationContext(out TOperationContext context))
+                {
+                    var json = await context.ReadForMemoryAsync(RequestHandler.RequestBodyStream(), "admin/revisions/delete");
+                    parameters = JsonDeserializationServer.Parameters.DeleteRevisionsParameters(json);
+                }
+
+                parameters.Validate();
 
                 using (var token = RequestHandler.CreateHttpRequestBoundTimeLimitedOperationToken())
                 {
-                    await DeleteRevisionsAsync(context, parameters.DocumentIds, includeForceCreated, token);
+                    deletedCount = await DeleteRevisionsAsync(parameters, token);
+                }
+
+                using (ContextPool.AllocateOperationContext(out TOperationContext context))
+                await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream()))
+                {
+                    writer.WriteStartObject();
+
+                    writer.WritePropertyName(nameof(DeleteRevisionsOperation.Result.TotalDeletes));
+                    writer.WriteInteger(deletedCount);
+
+                    writer.WriteEndObject();
+                }
+
+                throw new InvalidOperationException();
+            }
+            catch (Exception e)
+            {
+                ex = e;
+                throw;
+            }
+            finally
+            {
+                if (LoggingSource.AuditLog.IsInfoEnabled)
+                {
+                    RequestHandler.LogAuditFor(RequestHandler.DatabaseName, "DELETE", $"Delete revisions, totalDeletes: {deletedCount}" +
+                                                                                      (parameters == null ? string.Empty : $" ,parameters: {JsonSerializer.Serialize(parameters)} ") +
```

Comment:
`parameters` can be `null`? simply put here the `json` variable

Reply:
Done